### ISSUE TITLE
Issue with all_gather

### ISSUE THREAD

**Issue by scopello** [mlfoundations/open_clip#980]
Hi, 

`torch.distributed.nn.all_gather` was removed in recent pytorch versions (l'm using 2.4.0), which is used here: https://github.com/mlfoundations/open_clip/blob/49eac2f27a5bb98a7f7ecc1154918880aa55256c/src/open_clip/loss.py#L49

I am trying to use the recommended setting of `gather_with_grad=True` and `local_loss=True`. What is the best way to fix this? Thank you!

**Comment by rwightman** [comment]
@scopello sorry how is that deprecated? I'm using it in pytorch 2.5 right now...

**Comment by rwightman** [comment]
```python
import torch
torch.__version__
> '2.4.0'

import torch.distributed.nn

help(torch.distributed.nn.all_gather)

> Help on function all_gather in module torch.distributed.nn.functional:
> all_gather(tensor, group=None)
>     Gathers tensors from the whole group in a list.
>     Arguments:
>         tensor (Tensor): Tensor to be broadcast from current process.
>         group (ProcessGroup, optional): The process group to work on.
>     Returns:
>         tuple([Tensor]): Output of the collective.

```
